packages/strapi-design-system/src/DateTimePicker/DateTimePicker.js

```diff
@@ -0,0 +1,148 @@
+
+    let dateToSet;
+
+    if (timeValue) {
+      dateToSet = new Date(e);
+      dateToSet.setHours(timeValue.split(':')[0]);
+      dateToSet.setMinutes(timeValue.split(':')[1]);
+    } else {
+      dateToSet = new Date(e);
+      setTimeValue(`${dateToSet.getHours()}:${dateToSet.getMinutes()}:${dateToSet.getSeconds()}`);
+    }
+    if (onChange) {
+      onChange(dateToSet);
+    }
+  };
+
+  const handleTimeChange = (e) => {
+    setTimeValue(e);
+
+    const dateToSet = dateValue ? new Date(dateValue) : new Date();
+    dateToSet.setHours(e.split(':')[0]);
+    dateToSet.setMinutes(e.split(':')[1]);
+    if (!dateValue) {
+      setDateValue(dateToSet);
+    }
+
+    if (onChange) {
+      onChange(dateToSet);
+    }
+  };
+
+  const handleDateClear = () => {
+    setDateValue(undefined);
+    setTimeValue(undefined);
+    onClear();
+  };
+
+  const handleTimeClear = () => {
+    setTimeValue(undefined);
+
+    let dateToSet;
+
+    if (dateValue) {
+      dateToSet = new Date(dateValue);
+      dateToSet.setHours('00');
+      dateToSet.setMinutes('00');
+    }
+
+    if (onChange) {
+      onChange(dateToSet);
+    }
+  };
+
+  useEffect(() => {
+    if (value && +value !== +dateValue) {
+      const parsedData = parseDate(value);
+      setDateValue(parsedData);
+      setTimeValue(
+        parsedData ? `${parsedData.getHours()}:${parsedData.getMinutes()}:${parsedData.getSeconds()}` : null,
+      );
+    } else if (!value) {
+      setDateValue(undefined);
+      setTimeValue(undefined);
+    }
+  }, [value]);
+
+  const generatedId = useId('datetime-label', props?.id);
+
+  return (
+    <Field name={name} role="group" aria-labelledby={generatedId} hint={hint} error={error}>
+      <Stack spacing={1}>
+        {label && (
+          <FieldLabel required={required} action={labelAction} id={generatedId}>
+            {label}
+          </FieldLabel>
+        )}
+        <Stack horizontal spacing={2}>
+          <DatePicker
+            data-testid="datetimepicker-date"
+            name={name}
+            ariaLabel={label || ariaLabel}
+            error={typeof error === 'string'}
+            selectedDate={dateValue}
+            selectedDateLabel={(formattedDate) => `Date picker, current is ${formattedDate}`}
+            onChange={handleDateChange}
+            size={size}
+            onClear={onClear && handleDateClear}
+            clearLabel={clearLabel}
+            disabled={disabled}
+            {...props}
+          />
+          <TimePicker
+            data-testid="datetimepicker-time"
+            size={size}
+            aria-label={label || ariaLabel}
+            error={typeof error === 'string'}
+            value={timeValue}
+            onChange={handleTimeChange}
+            onClear={onClear && handleTimeClear}
+            clearLabel={clearLabel}
+            disabled={disabled}
+            step={step}
+          />
+        </Stack>
+        <FieldHint />
+        <FieldError id="datetimepicker" />
```

Comment:
This static ID needs to be replace with a dynamic one.

Reply:
done